test/plugins/test_autoimport.py

```diff
@@ -0,0 +1,236 @@
+    assert len(completions) > 0
+
+
+@pytest.mark.parametrize("completions", [("""def func(s:Lis""", 12)],
+                         indirect=True)
+def test_autoimport_function_typing(completions):
+    assert len(completions) > 0
+    assert check_dict({"label": "List"}, completions)
+
+
+@pytest.mark.parametrize("completions", [("""def func(s : Lis ):""", 16)],
+                         indirect=True)
+def test_autoimport_function_typing_complete(completions):
+    assert len(completions) > 0
+    assert check_dict({"label": "List"}, completions)
+
+
+@pytest.mark.parametrize("completions",
+                         [("""def func(s : Lis ) -> Generat:""", 29)],
+                         indirect=True)
+def test_autoimport_function_typing_return(completions):
+    assert len(completions) > 0
+    assert check_dict({"label": "Generator"}, completions)
+
+
+def test_autoimport_defined_name(config, workspace):
+    document = """List = "hi"\nLis"""
+    com_position = {"line": 1, "character": 3}
+    workspace.put_document(DOC_URI, source=document)
+    doc = workspace.get_document(DOC_URI)
+    completions = pylsp_autoimport_completions(config, workspace, doc,
+                                               com_position)
+    workspace.rm_document(DOC_URI)
+    assert not check_dict({"label": "List"}, completions)
+
+
+# This test has several large issues.
+# 1. autoimport relies on its sources being written to disk. This makes testing harder
+# 2. the hook doesn't handle removed files
+# 3. The testing framework cannot access the actual autoimport object so it cannot clear the cache
+# def test_autoimport_update_module(config: Config, workspace: Workspace):
+#     document2 = "SomethingYouShouldntWrite = 1"
+#     document = """SomethingYouShouldntWrit"""
+#     com_position = {
+#         "line": 0,
+#         "character": 3,
+#     }
+#     doc2_path = workspace.root_path + "/test_file_no_one_should_write_to.py"
+#     if os.path.exists(doc2_path):
+#         os.remove(doc2_path)
+#     DOC2_URI = uris.from_fs_path(doc2_path)
+#     workspace.put_document(DOC_URI, source=document)
+#     doc = workspace.get_document(DOC_URI)
+#     completions = pylsp_autoimport_completions(config, workspace, doc, com_position)
+#     assert len(completions) == 0
+#     with open(doc2_path, "w") as f:
+#         f.write(document2)
+#     workspace.put_document(DOC2_URI, source=document2)
+#     doc2 = workspace.get_document(DOC2_URI)
+#     pylsp_document_did_save(config, workspace, doc2)
+#     assert check_dict({"label": "SomethingYouShouldntWrite"}, completions)
+#     workspace.put_document(DOC2_URI, source="\n")
+#     doc2 = workspace.get_document(DOC2_URI)
+#     os.remove(doc2_path)
+#     pylsp_document_did_save(config, workspace, doc2)
+#     completions = pylsp_autoimport_completions(config, workspace, doc, com_position)
+#     assert len(completions) == 0
+#     workspace.rm_document(DOC_URI)
+
+
+class test_should_insert:
```

Comment:
```suggestion
class TestShouldInsert:
```

We prefer snake-cased names for classes.